kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -445,6 +447,12 @@ public void channelInactive(ChannelHandlerContext ctx) {
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         LOGGER.warn("Netty caught exception from the frontend: {}", cause.getMessage(), cause);
+        if (cause instanceof DecoderException de && de.getCause() instanceof FrameOversizedException e) {
```

Comment:
Why special case the exception handling here rather than just log this where we build the exception?

Reply:
I'd say separation of concerns.

Reply:
Separation of which concerns?

Reply:
The frame decoded doesn't know the context it's run in, just that it couldn't tolerate the bytes. It feels better to have the message decoder barf with exception because of the Frame size then have an exception handler that can produce a hint with context. Like figuring out if the virtual cluster has downstream TLS enabled, or if it's the first bytes being handled on the connection.

Reply:
Feels clunky but I'll live with it.